modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/util/MultiSelectItem.java

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.frontend.taglib.clay.servlet.taglib.util;
+
+import java.util.HashMap;
+
+/**
+ * @author Kresimir Coko
+ */
+public class MultiSelectItem extends HashMap<String, String> {
+
+	public MultiSelectItem() {
+	}
+
+	public MultiSelectItem(String label, String value) {
```

Comment:
I'm not sure we want to go with this overloading approach. I'd keep it simple following the same pattern present in `LabelItem` or `DropdownItem`.

Reply:
@carloslancha do you mean to say I should replace those 3 constructors with a single one that looks like this:
```java
public MultiSelectItem() {
	put("label", "label");
	put("value", "value");
}
```

That's the pattern I'm seeing in `LabelItem` and `DropdownItem`. Wouldn't that create an item with these fixed values instead of custom ones provided by the developer?

Reply:
Creating a constructor like

```
public MultiSelectItem() {
	put("label", "label");
	put("value", "value");
}
```

sets the default values. After creating an instance you can call `multiSelectItem.setLabel` or `multiSelectItem.setValue` to set your own if you want to customize them

And if you want to add other keys different from the default ones you can use `multiSelectItem.put("yourKey", "yourValue")` since this extends a `HashMap`.

Your approach is not wrong, I'm just suggesting to keep it as simple as possible. This is not being used anywhere so we can better add new API if requested than create it from the beginning.

Reply:
Oooh, I see, the extending part was missing from my understanding. Okay thank you @carloslancha , this clarifies it a lot!

Reply:
Hey @carloslancha, I'm having troubles with `multiSelectItem.put("yourKey", "yourValue")`.

Here's how I'm using it:
```java
MultiSelectItem _multiSelectItem1 = new MultiSelectItem();

_customSelectedItems.add(
	_multiSelectItem1.put(
		"name", "one"
	).put(
		"data", "1"
	).build()
);
```
I'm getting `cannot find symbol method put(String,String)` when building.

Reply:
First, use an instance of `MultiSelectItemList` instead an ArrayList, is the same but now that we have `MultiSelectItemList`... lets use it.

About the error, `MultiSelectItem` is not a Builder but a HashMap.

```
MultiSelectItem multiSelectItem = new MultiSelectItem();

multiSelectItem.put("name", "one");
multiSelectItem.put("data", "1");

MultiSelectItemList selectedItems = new MultiSelectItemList();

selectedItems.add(multiSelectItem);
```

Reply:
Hey @carloslancha I fixed this in [042cafb](https://github.com/liferay-frontend/liferay-portal/pull/528/commits/042cafb3a16e89451d9311fa6a06857170a60483) & [d22a629](https://github.com/liferay-frontend/liferay-portal/pull/528/commits/d22a629c1d795980274a55b6d0c47229460a54c5), with an SF commit on top because I didn't want to force push.

I didn't go with a builder/lambda approach.

Reply:
I'd say we need to go with builder to follow the same pattern we have for the rest of utils.

I'll take a look later and push it...

Reply:
Okay... Had I known earlier I'd have gone in that direction, but it was difficult for me to make sense of the lambda pattern when investigating this morning, so I went with the simple one.